### PR TITLE
add /_partition/partition/designdoc/ endpoints

### PR DESCRIPTION
**Pull request by garrensmith**
Adding partition endpoints along with adding some validation

### CODE REVIEW

File: src/chttpd/src/chttpd_db.erl [hunk 1]
```diff
@@ -250,24 +251,73 @@ handle_view_cleanup_req(Req, Db) ->
...
+            ok;
+        _ ->
+            throw({bad_request, <<"Partition is not allowed in the query string">>})
+    end.
+
+
 bad_action_req(#httpd{path_parts=[_, _, Name|FileNameParts]}=Req, Db, _DDoc) ->
     db_attachment_req(Req, Db, <<"_design/",Name/binary>>, FileNameParts).
 
+
```
Comment (rnewson) [inline src/chttpd/src/chttpd_db.erl]: ws only change.

File: src/chttpd/src/chttpd_db.erl [hunk 2]
```diff
@@ -250,24 +251,73 @@ handle_view_cleanup_req(Req, Db) ->
...
-    end;
+    DDoc = get_design_doc(DbName, Name),
+    Handler = chttpd_handlers:design_handler(Action, fun bad_action_req/3),
+    Handler(Req, Db, DDoc);
 
 handle_design_req(Req, Db) ->
     db_req(Req, Db).
 
+
+get_design_doc(DbName, Name) ->
```
Comment (rnewson) [inline src/chttpd/src/chttpd_db.erl]: the extraction of this function could be in a separate commit to make intention clearer. (You extracted it in order to use it from an extra location, rather than duplicating the code).

File: src/chttpd/src/chttpd_view.erl [hunk 3]
```diff
@@ -50,6 +54,7 @@ design_doc_view(Req, Db, DDoc, ViewName, Keys) ->
         fun couch_mrview_http:view_cb/2, VAcc, Args),
     {ok, Resp#vacc.resp}.
 
+
```
Comment (rnewson) [inline src/chttpd/src/chttpd_view.erl]: ws only change.

File: src/couch_mrview/src/couch_mrview_http.erl [hunk 4]
```diff
@@ -582,9 +582,6 @@ parse_param(Key, Val, Args, IsDecoded) ->
             Args#mrargs{callback=couch_util:to_binary(Val)};
         "sorted" ->
             Args#mrargs{sorted=parse_boolean(Val)};
-        "partition" ->
```
Comment (rnewson) [inline src/couch_mrview/src/couch_mrview_http.erl]: are we firmly against ?partition= working against _view? I quite like it for its simplicity.

Comment (garrensmith) [reply src/couch_mrview/src/couch_mrview_http.erl]: I think we against it otherwise its going to get really confusing having a ?partition and a new url

Comment (rnewson) [reply src/couch_mrview/src/couch_mrview_http.erl]: I would rather keep it. It's a feature of the map/reduce code and should be exposed there, rather than only reachable obscurely through mango.

**Comment by rnewson** [comment]
merged manually, closing.
